### DISCUSSION TITLE
Error 400 Strapi+Next

### DISCUSSION THREAD

**Discussion by dfdezhdez** [community/community#40050] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hola necesito ayuda, estoy intentando insertar datos desde Next hacia strapi 4 y recibo error 400, soy nueva haciendo mi primera app. El problema es que he seguido los pasos de video de Youtube y la version de strapi en el video es la 3. 

onst API_URL = process.env.NEXT_PUBLIC_API_URL || "http://localhost:1337/api/clientes";

export default function CrearNuevo() {
    
    
    const [values, setValues ] = useState({
        
           
            
     
        idcontrato:"",
           nombre:"",
        primer_apellido:"",
        segundo_apellido:"",
           
    
    });

   
    const { idcontrato, , nombre, primer_apellido, segundo_apellido} = values;
const router = useRouter

   const handleSubmit = async (e) => {
    e.preventDefault();

    
    
const emptyFieldCheck = Object.values(values).some(
    (element) => element === ""
);
if (emptyFieldCheck) {
    toast.error("Por favor rellena todos los campos");
}
const response = await fetch(`${API_URL}`, {
    method: "POST",
    headers: {
        "Content-Type": "application/json",
        
    },
    body: JSON.stringify(values),
    

});
if(!response.ok){
    toast.error("Revise, Algo anda mal")
} else {
    const cliente = await response.json();
    router.push(`/clientes/${cliente.slug}`);
}
};

   const handleInputChange = (e) => {
    const { name, value} = e.target;
    setValues({...values, [name]: value});
   };
   
   
   
    return(
        <LayoutClientes title="Crear Nuevo CLiente">
<Link href="/clientes/listarclientes">Ir Atras</Link>
<h2> Crear Nuevo CLiente</h2>
<ToastContainer />
<form onSubmit={handleSubmit} className={styles.form}>
    <div className={styles.grid}>
        
            
        <div>
            <label htmlFor="idcontrato">IDCONTRATO</label>
            <input
            name="idcontrato"
            type= "numeric"
            id= "idcontrato"
            value={idcontrato}
            onChange={handleInputChange}
            /> 
        </div>  
      
         <div>
            <label htmlFor="nombre">NOMBRE</label>
            <input
            name="nombre_cliente"
            type= "text"
            id= "nombre_cliente"
            value={nombre_cliente}
            onChange={handleInputChange}
            />     
        </div>
        <div>
            <label htmlFor="primer_apellido">PRIMER APELLIDO</label>
            <input
            name="primer_apellido"
            type= "text"
            id= "primer_apellido"
            value={primer_apellido}
            onChange={handleInputChange}
            />     
        </div>
        <div>
            <label htmlFor="segundo_apellido">SEGUNDO APELLIDO</label>
            <input
            name="segundo_apellido"
            type= "text"
            id= "segundo_apellido"
            value={segundo_apellido}
            onChange={handleInputChange}
            />     
        </div>
        <div>
            <la  <bel htmlFor="no_nucleo">NUCLEO</label>
            <input
            name="no_nucleo"
            type= "numeric"
            id= "no_nucleo"
            value={no_nucleo}
            onChange={handleInputChange}
            />     
        </div>
               
        </div>
        <input className="btn" type="submit" value="Crear"/>
   
        </form>
    </LayoutClientes>

    )

}

**Comment by ossanna16** [comment]
Hi @dfdezhdez,
Thanks for being a part of the GitHub Community! We're glad you're here.

I wanted to jump in and mention that if you're looking for help for this specific topic, you might want to try asking for help somewhere that focuses on Strapi + Next. It's definitely possible that another GitHub user might have run into this same issue and can help, but the GitHub Community on Discussions focuses primarily on topics related to GitHub itself or collaboration on project development and ideas. We want to make sure you’re getting the best support you can, but this space may not be the right place for this particular topic.

Best of luck!
